### PR TITLE
Add Symfony workflow configuration for Shipment

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13                   |
| Bug fix?        | no                                                       |
| New feature?    | yes                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/EventListener/Workflow/Shipment/ShipmentDateShipTransitionListener.php [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\EventListener\Workflow\Shipment;
+
+use Sylius\Bundle\ShippingBundle\Assigner\ShippingDateAssignerInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Symfony\Component\Workflow\Event\TransitionEvent;
+use Webmozart\Assert\Assert;
+
+final class ShipmentDateShipTransitionListener
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/EventListener/Workflow/Shipment/ShipmentDateShipTransitionListener.php]:
```suggestion
final class AssignShippingDateListener
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/order_shipping.xml [hunk 2]
```diff
@@ -16,11 +16,11 @@
         <defaults public="false" />
 
         <service id="Sylius\Bundle\CoreBundle\EventListener\Workflow\OrderShipping\AfterOrderShippedListener">
-            <argument type="service" id="sylius.state_resolver.order_shipping" />
+            <argument type="service" id="sylius.state_resolver.order" />
             <tag name="kernel.event_listener"
                  event="workflow.sylius_order_shipping.completed.ship"
                  method="onOrderShippingCompleted"
-                 priority="-100"
+                 priority="-200"
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/order_shipping.xml]: Why?

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/shipment.xml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+<container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+    <services>
+        <defaults public="false" />
+
+        <service id="Sylius\Bundle\CoreBundle\EventListener\Workflow\Shipment\ResolveOrderShipmentStateListener">
+            <argument type="service" id="sylius.state_resolver.order_shipping" />
+            <tag name="kernel.event_listener"
+                 event="workflow.sylius_shipment.completed.ship"
+                 priority="-200"
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/shipment.xml]:
```suggestion
                 priority="-100"
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/shipment.xml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            />
+        </service>
+
+        <service id="Sylius\Bundle\CoreBundle\EventListener\Workflow\Shipment\ShipmentDateShipTransitionListener">
+            <argument type="service" id="sylius.shipping_date_assigner" />
+            <tag name="kernel.event_listener"
+                 event="workflow.sylius_shipment.transition.ship"
+                 priority="-200"
```
Comment (GSadee) [inline src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/shipment.xml]:
```suggestion
                 priority="-100"
```

File: src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/order_shipping.xml (hunk 2 above)
Comment (TheMilek) [reply src/Sylius/Bundle/CoreBundle/Resources/config/services/listeners/workflow/order_shipping.xml]: copied and forgot to change it 😅

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Kamil! :1st_place_medal:

**Pull request merged**